packages/backend/scripts/createOpStack.ts

```diff
@@ -0,0 +1,214 @@
+import { mkdir, readFile, writeFile } from 'fs/promises'
+import path from 'path'
+import util from 'util'
+
+const execPromise = util.promisify(exec)
+
+interface HelpMode {
+  mode: 'help'
+}
+
+interface OpStackMode {
+  mode: 'opStack'
+  project: string
+  initialAddresses: [EthereumAddress, EthereumAddress]
+}
+
+type Config = HelpMode | OpStackMode
+
+void main().catch((e) => {
+  console.error(e)
+  process.exit(1)
+})
+
+async function main() {
+  const config = parseCliParameters()
+  switch (config.mode) {
+    case 'help':
+      printUsage()
+      break
+    case 'opStack':
+      await createOpStackProject(config.project, config.initialAddresses)
+      break
+  }
+}
+
+async function createOpStackProject(
+  project: string,
+  initialAddresses: EthereumAddress[],
+): Promise<void> {
+  const dir = path.join(__dirname, '..', 'discovery', project, 'ethereum')
+  await mkdir(dir, { recursive: true })
+
+  let projectConfig: RawDiscoveryConfig = configTemplate(
+    project,
+    initialAddresses,
+    names,
+  )
+
+  console.log('Running initial discovery for', project)
+  await writeFile(
+    path.join(dir, 'config.jsonc'),
+    JSON.stringify(projectConfig, null, 2),
+  )
+  console.log('Initial discovery...')
+  await execPromise('yarn discover ethereum ' + project, { cwd: __dirname })
+  console.log('Initial discovery done!')
+
+  let i = 1
+  const maxTries = 5
+  while (
+    Object.keys(projectConfig.names ?? {}).length !== NAMES.length &&
+    i < maxTries
+  ) {
+    console.log('Backfilling addresses...')
+    projectConfig = await backfillAddresses(dir, projectConfig)
+    await writeFile(
+      path.join(dir, 'config.jsonc'),
+      JSON.stringify(projectConfig, null, 2),
+    )
+    console.log('Running discovery:', i++)
+    await execPromise('yarn discover ethereum ' + project, { cwd: __dirname })
+  }
+
+  console.log('Prettyfying config...')
+  await execPromise(
+    `yarn format:fix discovery/${project}/ethereum/config.jsonc`,
+  )
+
+  if (i === maxTries) {
+    console.error('Failed to backfill all addresses')
+  }
+
+  console.log('Done! Checkout the config at')
+  console.log(path.join(dir, 'config.jsonc'))
+  console.log('and the discovery output at')
+  console.log(path.join(dir, 'discovered.json'))
+}
+
+async function backfillAddresses(dir: string, oldConfig: RawDiscoveryConfig) {
+  const discoveryOutput = await readFile(path.join(dir, 'discovered.json'))
+  const discovery = JSON.parse(discoveryOutput.toString()) as {
+    contracts: { name: string; address: string }[]
+  }
+
+  const names = discovery.contracts.reduce((acc, contract) => {
+    if (NAMES.includes(contract.name)) {
+      acc[contract.address] = contract.name
+    }
+    return acc
+  }, {} as Record<string, string>)
+
+  assert(typeof oldConfig === 'object', 'oldConfig must be an object')
+  return configTemplate(oldConfig.name, oldConfig.initialAddresses, names)
+}
+
+function parseCliParameters(): Config {
```

Comment:
This should take the name of the stack as a subcommand, so we call:

- `yarn init-project opstack 0xabcd`
- `yarn init-project orbitstack 0xabcd`